ffi/derive/src/ffi_fn.rs

```diff
@@ -185,10 +205,11 @@ fn gen_method_call_stmt(fn_descriptor: &FnDescriptor) -> TokenStream {
 fn gen_output_assignment_stmts(fn_descriptor: &FnDescriptor) -> TokenStream {
     if let Some(output_arg) = &fn_descriptor.output_arg {
         if let Some(receiver) = &fn_descriptor.receiver {
-            let arg_name = receiver.name();
-            let src_type = receiver.src_type();
-
-            if matches!(src_type, syn::Type::Path(_)) {
+            // NOTE: case like fn(self, ...) -> Self (builder pattern)
+            if matches!(receiver.src_type(), syn::Type::Path(_))
+                && receiver.name() == output_arg.name()
```

Comment:
```suggestion
            if receiver.name() == output_arg.name()
```

again, this should be sufficient. Would you extract it into a function and add a description, this is very succint. Both here and earlier

Reply:
maybe you can take out this whole if nesting into a function

Reply:
@mversic, fixed, rewrite function a bit to avoid nesting